File: senseiVoiceInterface/src/app/services/voix.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

export interface VoixResponse {
  uuid: string;
  code: string;
  speakerUuid: string;
  nombrePoint: number;
  gender: string;
  language: string;
  type: string;
  name: string;
  price: number;
  url: UrlInfo;
}

export interface UrlInfo {
  id: string;
  name: string;
  type: string;
  voice_engine: string;
}

export interface VoixRequest {
  name: string;
  gender: string;
  language: string;
  price: string;
}

@Injectable({
  providedIn: 'root'
})
export class VoixService {
  private apiUrl = `${environment.apiUrl}/api/voix`;

  constructor(private http: HttpClient) {}

  createVoice(file: File, voiceName: string, userId: number): Observable<VoixResponse> {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('voiceName', voiceName);
    formData.append('id', userId.toString());
    return this.http.post<VoixResponse>(`${this.apiUrl}/create`, formData);
  }

  getVoicesBySpeaker(uuid: string | null): Observable<VoixResponse[]> {
    return this.http.get<VoixResponse[]>(`${this.apiUrl}/speaker-voice/${uuid}`);
  }
  getVoiceById(id: string): Observable<VoixResponse> {
    return this.http.get<VoixResponse>(`${this.apiUrl}/${id}`);
  }

  getAllVoices(): Observable<VoixResponse[]> {
    return this.http.get<VoixResponse[]>(`${this.apiUrl}/all`);
  }

  updateVoice(id: number, voixRequest: VoixRequest): Observable<VoixResponse> {
    return this.http.put<VoixResponse>(`${this.apiUrl}/update/${id}`, voixRequest);
  }

  deleteVoice(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/delete/${id}`);
  }
}

File: voiceOverFinal2/src/main/java/com/example/senseiVoix/services/VoixService.java
package com.example.senseiVoix.services;
import com.example.senseiVoix.dtos.voix.VoixRequest;
import com.example.senseiVoix.entities.Utilisateur;
import com.example.senseiVoix.entities.Voix;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;
import java.util.Optional;

public interface VoixService {

    Voix createVoice(MultipartFile file, String voiceName, Utilisateur user, String gender, String language, Double price) throws IOException;

    Optional<Voix> getVoiceById(Long id);

    List<Voix> getAllVoices();

    List<Voix> findAllBySpeaker(String uuid);



    Voix findByUuid(String id);

    List<Voix> findAll();

    Voix updateVoice(Long id, VoixRequest voixRequest);

    boolean deleteVoice(Long id);
}
